src/AltinnCore/Runtime/RestControllers/DataController.cs

```diff
@@ -18,32 +35,123 @@ public class DataController : ControllerBase
         private readonly IData dataService;
         private readonly IInstance instanceService;
 
+        private readonly IExecution executionService;
+        private readonly UserHelper userHelper;
+        private readonly IPlatformServices platformService;
+        private readonly IRepository repositoryService;
+        private readonly IInstanceEvent eventService;
+
+        private const long REQUEST_SIZE_LIMIT = 2000 * 1024 * 1024;
+
         /// <summary>
-        /// The data controller. Currently only GET is supported.
+        /// The data controller is responsible for adding business logic to the data elements.
         /// </summary>
+        /// <param name="generalSettings">settings </param>
         /// <param name="logger">logger</param>
+        /// <param name="registerService">register service</param>
+        /// <param name="instanceService">instance service to store instances</param>
         /// <param name="dataService">dataservice</param>
+        /// <param name="executionService">execution service to execute data element logic</param>
+        /// <param name="profileService">profile service to access profile information about users and parties</param>
+        /// <param name="platformService">platform</param>
+        /// <param name="repositoryService">repository for accessing applicaiton metadata</param>
+        /// <param name="eventService">event service for dispatching data events</param>
         public DataController(
+            IOptions<GeneralSettings> generalSettings,
             ILogger<DataController> logger,
+            IRegister registerService,
+            IInstance instanceService,
             IData dataService,
-            IInstance instanceService)
+            IExecution executionService,
+            IProfile profileService,
+            IPlatformServices platformService,
+            IRepository repositoryService,
+            IInstanceEvent eventService)
         {
             this.logger = logger;
-            this.dataService = dataService;
+
             this.instanceService = instanceService;
+            this.dataService = dataService;
+            this.executionService = executionService;
+            this.platformService = platformService;
+            this.repositoryService = repositoryService;
+            this.eventService = eventService;
+            this.userHelper = new UserHelper(profileService, registerService, generalSettings);
+        }
+
+        /// <summary>
+        ///  Gets a data element from storage and performs business logic on it (e.g. to calculate certain fields) before it is returned.
+        ///  If more there are more data elements of the same elementType only the first one is returned. In that case use the more spesific
+        ///  GET method to fetch a particular data element. 
+        /// </summary>
+        /// <param name="org">uniqe identfier of the organization responsible for the app</param>
```

Comment:
"...ident**i**fier..." and "...organi**s**ation..." ("s" as you used below and for future use in the codebase.)